### ISSUE TITLE
Users unable to confirm their email addresses

### ISSUE THREAD

**Issue by henrywright** [julianlam/nodebb-plugin-sso-twitter#13]
The plugin asks users to verify their email address after registration but no email is sent out. Looking at the member's profile I can see their email is "undefined".

**Comment by julianlam** [comment]
Thanks, this might be a core issue as Twitter wouldn't be the only SSO plugin that would register users without emails. 

I'll look into this.

**Comment by henrywright** [comment]
I'm not entirely sure if Twitter will give out user's email addresses via their API. Perhaps that's why we're seeing the "undefined' value?

**Comment by julianlam** [comment]
For sure they do not. Unless something has changed in recent years.

**Comment by henrywright** [comment]
Right. In that case, there is an alert that pops up asking a new user to click here to confirm their email. After "clicking here" the user is told the email has been sent. In fact it hasn't because there's no email to send to. 

Perhaps the solution is to ask the user to provide their email address before anything else is done?

**Comment by julianlam** [comment]
In core, we should just not bother with that alert. 

If you use Twitter login solely, you should turn off the option that restricts posting to validated users.

**Comment by henrywright** [comment]
> If you use Twitter login solely, you should turn off the option that restricts posting to validated users.

OK that makes sense but my problem is I allow email registration too. Turning off email validation worries me :(

**Comment by julianlam** [comment]
Instructing users to add their email is an acceptable workaround... I'll do
that.

**Comment by Fastidious** [comment]
I am having the same problem. With the Google SSO, there is an option not to require email verification, perhaps this could be implemented here? I am noticing that Google and Twitter SSO bypass "Admin Approval" on Authentication. Problem too?

**Comment by Giggiux** [comment]
Can you implement the  "not require email verification" here too? (If possible also on Steam)

Thanks!
